src/jit/hwintrinsicxarch.cpp

```diff
@@ -136,6 +139,77 @@ NamedIntrinsic Compiler::lookupHWIntrinsic(const char* methodName, InstructionSe
     return result;
 }
 
+//------------------------------------------------------------------------
+// insOfHWIntrinsic: map named intrinsic value to its corresponding instruction
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    baseType  -- the base type of the intrinsic (flt, dbl, i8, i16, i32, i64)
+//
+// Return Value:
+//    instruction for the intrinsic
+//
+instruction Compiler::insOfHWIntrinsic(NamedIntrinsic intrinsicID, var_types baseType)
+{
+    assert(intrinsicID != NI_Illegal);
+    assert(intrinsicID > NI_HW_INTRINSIC_START && intrinsicID < NI_HW_INTRINSIC_END);
+
+    int index = -1;
+
+    switch (baseType)
+    {
+        case TYP_BYTE:
+        case TYP_UBYTE:
+            index = 2;
+            break;
+
+        case TYP_SHORT:
+        case TYP_USHORT:
+            index = 3;
+            break;
+
+        case TYP_INT:
+        case TYP_UINT:
+            index = 4;
+            break;
+
+        case TYP_LONG:
+        case TYP_ULONG:
+            index = 5;
+            break;
+
+        case TYP_FLOAT:
+            index = 0;
+            break;
+
+        case TYP_DOUBLE:
+            index = 1;
+            break;
+
+        default:
+            unreached();
+            break;
+    }
+
+    return hwIntrinsicInfoArray[intrinsicID - NI_HW_INTRINSIC_START - 1].ins[index];
+}
+
+//------------------------------------------------------------------------
+// ivalOfHWIntrinsic: map named intrinsic value to its corresponding ival
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//
+// Return Value:
+//    ival for the intrinsic
+//
+ssize_t Compiler::ivalOfHWIntrinsic(NamedIntrinsic intrinsicID)
```

Comment:
Where's this "ival" thing used?

Reply:
In generating instructions - immediate bytes are a part of the instruction encoding

Reply:
I suspect that but I don't see it used anywhere in this PR. It's problematic to review code that it isn't used.

Reply:
There are 2, possibly 3, SxS PRs going on that will need to use this method. It exists so that we don't hit merge conflicts.

I can give examples and link to one of the existing PRs if that helps.

Reply:
I think the main question is if this belongs in Compiler or in CodeGen. I suspect that it will only be needed in CodeGen so...

Reply:
Working on an update to move it into CodeGen and make it static now.